internal/services/providers/eks/eks.go

```diff
@@ -151,3 +145,35 @@ func (p *Provider) FilterSpot(ctx context.Context, nodes []*v1.Node) ([]*v1.Node
 func (p *Provider) Name() string {
 	return Name
 }
+
+func isLabeledSpot(n *v1.Node) bool {
```

Comment:
instead of "isLabeledSpot/isLabeledOnDemand", create a function "determineLifecycle", with response type "spot", "on-demand" or "i don't know". will help avoid some of the duplication.